### PR TITLE
Math magicians: components

### PR DESCRIPTION
**Pull request by bhushan354**
In this PR I have implemented the following changes :

- Deleted all the boilerplate from Create React App (text, images, styles).
- Created a new directory called components.
- Created a simple looking calculatorUi using JSX , and basic css .
- Also added some design of my choice .

Thank you ,
Bhushan .

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
+  render(<App />);
+  const linkElement = screen.getByText(/learn react/i);
+  expect(linkElement).toBeInTheDocument();
+});
```
Comment (Johnadibe) [inline src/App.test.js]: > - [x] According to the project requirement, you should remove all the default files generated by Create React App (CRA). Please go ahead and delete this file, as it's not needed for this project. This action will help you align with the project's requirements. 👍

File: src/reportWebVitals.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      getCLS(onPerfEntry);
+      getFID(onPerfEntry);
+      getFCP(onPerfEntry);
+      getLCP(onPerfEntry);
+      getTTFB(onPerfEntry);
+    });
+  }
+};
+
+export default reportWebVitals;
```
Comment (Johnadibe) [inline src/reportWebVitals.js]: > - [x] According to the project requirement, you should remove all the default files generated by Create React App (CRA). Please go ahead and delete this file, as it's not needed for this project. This action will help you align with the project's requirements. 👍

File: src/setupTests.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom';
```
Comment (Johnadibe) [inline src/setupTests.js]: > - [x] According to the project requirement, you should remove all the default files generated by Create React App (CRA). Please go ahead and delete this file, as it's not needed for this project. This action will help you align with the project's requirements. 👍

File: src/components/Calculator.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <div className="calciGrid">
+      <div className="output">
+        <div className="prev-Num" />
+        <div className="curr-Num">0</div>
+      </div>
+      <button type="button">AC</button>
+      <button type="button">+/-</button>
+      <button type="button">%</button>
+      <button type="button" className="orngBtn">+</button>
```
Comment (Johnadibe) [inline src/components/Calculator.js]: > - [x] Based on the design, this should be a `÷` and not `+`. Kindly implement that👍
> 
> Provided design:
> 
> ok11
> 
> Your design:
> 
> ok1
>

File: src/components/Calculator.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+export default function Calculator() {
+  return (
+    <div className="calciGrid">
+      <div className="output">
+        <div className="prev-Num" />
+        <div className="curr-Num">0</div>
+      </div>
```
Comment (Johnadibe) [inline src/components/Calculator.js]: > - [ ] In accordance with the project requirements, it is advisable to introduce a child component within the Calculator.js file. I recommend the extraction of this code into a separate component, which can subsequently be called within the Calculator.js file. By encapsulating specific functionality or UI elements into child components, it becomes easier to manage and reuse code, facilitating collaboration and future updates.😊
> 
> ```
> // Calculator.js (Parent Component)
> import React from 'react';
> import CalculatorDisplay from './CalculatorDisplay'; // Import the child component
> 
> export default function Calculator() {
>   return (
>     <div className="calciGrid">
>       {/* Render the child component */}
>       <CalculatorDisplay />
>       {/* Other calculator buttons and UI elements */}
>     </div>
>   );
> }
> ```
> 
> 
> ```
> // CalculatorDisplay.js (Child Component)
> import React from 'react';
> 
> function CalculatorDisplay() {
>   return (
>     <div className="output">
>         <div className="prev-Num" />
>         <div className="curr-Num">0</div>
>       </div>
>   );
> }
> 
> export default CalculatorDisplay;
> 
> ```

**Review by Johnadibe — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️
### Hi @bhushan354👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

mario_ladders

## Highlights
✅ You used GitFlow correctly 👍🏽
✅ Linter checks passed✔️
✅ Created a new directory called Components✔️
✅ Created a new Calculator.js file Inside the components directory✔️
✅ Used functional components✔️


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ademibowale — APPROVED** [review]
Hi @bhushan354  , 


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️
- Use functional components✔️
- Delete all the boilerplate from Create React App (text, images, styles).✔️
- The project have a directory called components ✔️
- Descriptive commit messages ✔️ 
- Good use of Github flow ✔️






- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**